roles/satellite-clone/tasks/main.yml

```diff
@@ -184,6 +184,16 @@
     state: latest
   when: puppet_version|int == 4
 
+  # Because of https://projects.theforeman.org/issues/30506
+- name: Install Satellite SELinux packages for 6.8 and above
```

Comment:
There's a lot of checks in this module relating to the `satellite_version`. A more flexible and generic way would be to define an array in the respective vars files called something like `selinux_packages` which would be empty for versions other than 6.8. That means this task would be more along the lines of "install any specific selinux packages" and we don't care about satellite version at all. That also gives us more fine-grained control per satellite version!

Ultimately I'd love to see us move in that direction so that the playbooks become quite a bit 'dumber'

Reply:
The pattern I suggest is also an improvement because when it's time to support 6.9 and we copy the 6.8 vars yaml we can review it (perhaps adding more -selinux packages for example) to ensure we don't miss anything. The way this is currently written would require review of this playbook and hopefully not missing this

Reply:
I like the idea

Reply:
Thought about it some more. In this particular case maybe it's fair enough to just have it ensure these packages are installed for any version. Downside there would be if the list should actually change between versions in the future. then we'd need something like I suggested above

Reply:
hey @jturel do you mind creating a github issue to think about it in future, I do like the idea of keeping the version-specific changes under control and separate.

Reply:
Done! https://github.com/RedHatSatellite/satellite-clone/issues/383

Reply:
It looks like in 6.7 we do not have a crane package, so the job is failing with:

```bash
2021-01-11 20:07:02,915 p=15208 u=root n=ansible | fatal: [localhost]: FAILED! => {"changed": false, "msg": "No package matching 'crane-selinux' found available, installed or updated", "rc": 126, "results": ["No package matching 'crane-selinux' found available, installed or updated"]}
```

Looking at a 6.7 box all I see for crane is:

```bash
[root@satellite1 ~]# rpm -qa | grep crane
python2-crane-3.3.1-9.el7sat.noarch
```

Should we change this to just run on 6.8 or go with the var defined in the yaml file option?

Reply:
There are two options imo:

1. Write this command such that it won't fail the ansible run. Probably not the best solution but I suppose it would work
2. Yep. Do what I outlined here https://github.com/RedHatSatellite/satellite-clone/issues/383

Reply:
> 2\. Yep. Do what I outlined here #383

+1, you can make the default a blank array [here](https://github.com/RedHatSatellite/satellite-clone/blob/master/roles/satellite-clone/defaults/main.yml) and set it to the expected packages [here](https://github.com/RedHatSatellite/satellite-clone/blob/master/roles/satellite-clone/vars/satellite_6.8.yml)

It looks like you can use a comma separated list of packages with the [yum module](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/yum_module.html)

>To operate on several packages this can accept a comma separated string of packages or (as of 2.0) a list of packages.

So you may just be able to pass the variable and it will work fine, or can join with a filter e.g. `{{ selinux_pkgs | join(",") }}`

You will probably need to skip the step for a blank array, so you could do something like `when: selinux_pkgs | length > 0` iirc